codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/eventstream/ServerEventStreamUnmarshallerGeneratorTest.kt

```diff
@@ -7,67 +7,36 @@ package software.amazon.smithy.rust.codegen.server.smithy.protocols.eventstream
+import software.amazon.smithy.rust.codegen.core.testutil.IntegrationTestParams
+import software.amazon.smithy.rust.codegen.core.testutil.integrationTest
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverIntegrationTest
 
 class ServerEventStreamUnmarshallerGeneratorTest {
     @ParameterizedTest
     @ArgumentsSource(TestCasesProvider::class)
-    fun test(testCase: TestCase) {
+    fun some_tests(testCase: TestCase) {
         // TODO(https://github.com/awslabs/smithy-rs/issues/1442): Enable tests for `publicConstrainedTypes = false`
         // by deleting this if/return
         if (!testCase.publicConstrainedTypes) {
             return
         }
 
-        EventStreamTestTools.runTestCase(
-            testCase.eventStreamTestCase,
-            object : ServerEventStreamBaseRequirements() {
-                override val publicConstrainedTypes: Boolean get() = testCase.publicConstrainedTypes
+        serverIntegrationTest(
+            testCase.eventStreamTestCase.model,
+            IntegrationTestParams(service = "test#TestService", addModuleToEventStreamAllowList = true),
+        ) { codegenContext, rustCrate ->
+            val crateName = codegenContext.moduleUseName()
+            val generator = "$crateName::event_stream_serde::TestStreamUnmarshaller"
 
-                override fun renderGenerator(
-                    codegenContext: ServerCodegenContext,
-                    project: TestEventStreamProject,
-                    protocol: Protocol,
-                ): RuntimeType {
-                    fun builderSymbol(shape: StructureShape): Symbol = shape.serverBuilderSymbol(codegenContext)
-                    return EventStreamUnmarshallerGenerator(
-                        protocol,
-                        codegenContext,
-                        project.operationShape,
-                        project.streamShape,
-                        ::builderSymbol,
-                    ).render()
-                }
-
-                // TODO(https://github.com/awslabs/smithy-rs/issues/1442): Delete this function override to use the correct builder from the parent class
-                override fun renderBuilderForShape(
-                    writer: RustWriter,
-                    codegenContext: ServerCodegenContext,
-                    shape: StructureShape,
-                ) {
-                    BuilderGenerator(codegenContext.model, codegenContext.symbolProvider, shape, emptyList()).apply {
-                        render(writer)
-                        writer.implBlock(codegenContext.symbolProvider.toSymbol(shape)) {
-                            renderConvenienceMethod(writer)
-                        }
-                    }
-                }
-            },
-            CodegenTarget.SERVER,
-            EventStreamTestVariety.Unmarshall,
-        )
+            rustCrate.integrationTest("unmarshall") {
```

Comment:
You don't have to use `rustCrate.integrationTest`. This is probably what's requiring you to make modules public, as this puts tests in the `tests/` directory. Rather, you can put them directly inline in whatever module you want to test with `rustCrate.moduleFor` and just manually add the `cfg(test)` and `#[tokio::test]` parts (or there might even be a `unitTest` function somewhere).

Reply:
rustCrate.testModule works best here